<?xml version="1.0"?>
<launch>
  <arg name="world" default="RMUL2024_world" doc="model type [RMUC2024_world, RMUL2024_world, ...]" />
  <arg name="robot_x" default="4.75" />
  <arg name="robot_y" default="4.54" />
  <arg name="robot_z" default="1.159720" />
  <!-- <arg name="robot_x" default="1" />
  <arg name="robot_y" default="0" />
  <arg name="robot_z" default="0.2" /> -->
  <arg name="use_sim" default="true"/>


  <arg name="local_planner" default="mpc" doc="controller type [mpc, dwa, ...]" />

  <param name="/use_sim_time" value="true"/>
  <!-- 是否启动gazebo仿真 -->

  <node name="map_server" pkg="map_server" type="map_server" args="$(find rm_local_planner)/map/$(arg world)/$(arg world).yaml" output="screen">
  <!-- <node name="map_server" pkg="map_server" type="map_server" args="$(find rm_local_planner)/map/RMUL/RMUL.yaml" output="screen"> -->
    <param name="frame_id" value="map" />
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rm_local_planner)/config/rviz/rm_local_planner.rviz"/>

  <include file="$(find rm_local_planner)/launch/amcl_simple_meca_car.launch"/>

  <!-- <include file="$(find scan_to_map)/launch/scan_to_map_location.launch"/> -->

    <!--启动仿真-->
  <include file="$(find simple_meca_car)/launch/gazebo.launch" if="$(eval arg('use_sim')==true)">
      <arg name="world"   value="$(arg world)" />
      <arg name="robot_x" value="$(arg robot_x)" />
      <arg name="robot_y" value="$(arg robot_y)" />
      <arg name="robot_z" value="$(arg robot_z)" />
  </include>

  <!--Launch the move base with time elastic band-->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find rm_local_planner)/config/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find rm_local_planner)/config/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find rm_local_planner)/config/params/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find rm_local_planner)/config/params/global_costmap_params.yaml" command="load" />
    
    <param name="base_global_planner" value="global_planner/GlobalPlanner" />
    <!--<param name="neutral_cost" value="220" />-->
    <param name="planner_frequency" value="3.0" />
    <param name="planner_patience" value="5.0" />
    <!-- 默认为A*算法 -->
    <param name="use_dijkstra" value="false" />
    
    <!-- <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <param name="controller_frequency" value="20.0" />
    <param name="controller_patience" value="15.0" /> -->
    <!-- <param name="base_local_planner" value="mpc_planner/MPCPlanner" if="$(eval arg('local_planner')=='mpc')" />
        <rosparam file="$(find rm_local_planner)/config/params/mpc_planner_params.yaml" command="load" if="$(eval arg('local_planner')=='mpc')" /> -->


    <param name="clearing_rotation_allowed" value="false" />
    <remap from="/cmd_vel" to="/" />
  </node>

  <include file="$(find rm_local_planner)/launch/control_base.launch">
      <arg name="controller_type" value="mpc"/>
      <arg name="mpc_speed_value" value="8"/>
      <arg name="max_x_speed" default="1" />
      <arg name="max_y_speed" default="1" />
      <arg name="p_value" default="1.5" />
      <arg name="i_value" default="1.0"/>
      <arg name="d_value" default="1.0"/>
  </include> 
  <!-- <include file="$(find rm_decision)/launch/rm_decision.launch"/> -->


</launch>
